sdk/storage/azure-storage-files-shares/inc/azure/storage/files/shares/share_responses.hpp

```diff
@@ -190,6 +190,43 @@ namespace Azure { namespace Storage { namespace Files { namespace Shares {
       DateTime LastModified;
     };
 
+    enum class ShareFileHandleAccessRights
```

Comment:
Why don't we make this an extensible enum, just like `FileAttributes`?

Reply:
Because I think it is more easily used by customers as a flag enum, like `ListSharesIncludeFlags`.
It is also a flag enum in [dotnet SDK](https://github.com/Azure/azure-sdk-for-net/pull/34512/files#diff-62b406a22820e487a420f2d8573657384bba1139fa517054728f75536c4c426a).
I think  whether an enum should be an extensible enum is whether it allows values other than the enum value.
And I referenced to this about [extensible enum](https://stackoverflow.com/questions/12118654/how-to-define-an-extensible-c-enum-system)

Reply:
And I'd like to know what is the reason you want it to be an extensible enum?

Reply:
Extendable enumerations have the ability to be expanded over time. C++ Enumerations are fixed for all time (or more specifically, changing a C++ enumeration is considered a breaking change).

Reply:
I got it. I will change this to extendable enumeration. Thanks!